core/src/integration-test/java/com/scalar/db/storage/jdbc/JdbcAdminImportTestUtils.java

```diff
@@ -435,4 +393,165 @@ private String prepareCreateTableSql(
+        SUPPORTED_TABLE_NAME + "_long_raw",
+        prepareCreateTableSql(
+            namespace,
+            SUPPORTED_TABLE_NAME + "_long_raw",
+            prepareColumnsForOracleLongRaw(),
+            longRawTableMetadata.getPartitionKeyNames()));
+    supportedTableMetadata.put(SUPPORTED_TABLE_NAME + "_long_raw", longRawTableMetadata);
+
+    Map<String, String> unsupportedTables =
+        prepareCreateNonImportableTableSql(
+            namespace,
+            majorVersion >= 20
+                ? Stream.concat(
+                        UNSUPPORTED_DATA_TYPES_ORACLE.stream(),
+                        UNSUPPORTED_DATA_TYPES_ORACLE_V20_OR_LATER.stream())
+                    .collect(Collectors.toList())
+                : UNSUPPORTED_DATA_TYPES_ORACLE);
+
+    return executeCreateTableSql(supportedTables, supportedTableMetadata, unsupportedTables);
+  }
+
+  private Map<String, TableMetadata> createExistingSqlServerDatabaseWithAllDataTypes(
+      String namespace) throws SQLException {
+    TableMetadata tableMetadata = prepareTableMetadataForSqlServer();
+    Map<String, String> supportedTables =
+        Collections.singletonMap(
+            SUPPORTED_TABLE_NAME,
+            prepareCreateTableSql(
+                namespace,
+                SUPPORTED_TABLE_NAME,
+                prepareColumnsForSqlServer(),
+                tableMetadata.getPartitionKeyNames()));
+    Map<String, TableMetadata> supportedTableMetadata =
+        Collections.singletonMap(SUPPORTED_TABLE_NAME, tableMetadata);
+
+    Map<String, String> unsupportedTables =
+        prepareCreateNonImportableTableSql(namespace, UNSUPPORTED_DATA_TYPES_MSSQL);
+
+    return executeCreateTableSql(supportedTables, supportedTableMetadata, unsupportedTables);
+  }
+
+  private Map<String, TableMetadata> executeCreateTableSql(
+      Map<String, String> supportedTables,
+      Map<String, TableMetadata> supportedTableMetadata,
+      Map<String, String> unsupportedTables)
+      throws SQLException {
+    Map<String, TableMetadata> results = new HashMap<>();
+    List<String> sqls = new ArrayList<>();
+
+    // table with all supported columns
+    supportedTables.forEach(
+        (table, sql) -> {
+          sqls.add(sql);
+          results.put(table, supportedTableMetadata.get(table));
+        });
+
+    // tables with an unsupported column
+    unsupportedTables.forEach(
+        (table, sql) -> {
+          sqls.add(sql);
+          results.put(table, null);
+        });
+
+    execute(sqls.toArray(new String[0]));
+    return results;
+  }
+
+  private boolean isMariaDB() {
+    try (BasicDataSource dataSource = JdbcUtils.initDataSourceForAdmin(config, rdbEngine);
+        Connection connection = dataSource.getConnection()) {
+      String version = connection.getMetaData().getDatabaseProductVersion();
+      return version.contains("MariaDB");
+    } catch (SQLException e) {
+      throw new RuntimeException("Get database major version failed");
```

Comment:
```suggestion
      throw new RuntimeException("Get database product version failed");
```
I missed that.

Reply:
Good catch! Thank you! Fixed in 7cb850a.